### ISSUE TITLE
polymer-dart pub build not supported behind proxy

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#729]
_Originally opened as dart-lang/sdk#15161_

*This issue was originally filed by empyrean...&#064;gmail.com*
_____
1. Create a new polymer library web application in DartEditor with sample content
2. either Build Polymer App from DartEditor (pub build --no-minify) or run 'pub build' from the command line

When running outside of a proxy, the build goal succeeds. When behind a corporate proxy, 'pub build' fails with an error: &quot;Transformer library &quot;package:polymer/transformer.dart&quot; not found.&quot; 

Dart 1.0.0.3_r30188, Mac OS X 10.8 and 10.9

Pub is able to see the proxy settings (&quot;pub upgrade&quot; successfully downloads dependencies) but the transformer library does not seem to respect those settings. Could this be related in any way to the IPv6 issues in [Issue dart-lang/pub#696]?  (though I'm not trying to run pub serve)

**Comment by DartBot** [comment]
*This comment was originally written by c.targett...&#064;gmail.com*
_____
I have encountered the same issue.

Pub files and output: https://gist.github.com/xlevus/acba5f930214eff6d693

Additionally, when running `pub build` with blank http_proxy environment variables, it also fails.

&nbsp;&nbsp;&nbsp;&nbsp;$ http_proxy='' https_proxy='' pub build
&nbsp;&nbsp;&nbsp;&nbsp;Transformer library "package:polymer/transformer.dart" not found.

While

&nbsp;&nbsp;&nbsp;&nbsp;$ env --unset=http_proxy --unset=https_proxy pub build

Works.

**Comment by DartBot** [comment]
*This comment was originally written by c.targ...&#064;gmail.com*
_____
And further, setting the `no_proxy` env var to contain IPv6 localhost addresses, does not work either.

$ no_proxy="localhost,127.0.0.0/8,::1,[::1]" pub build
Transformer library "package:polymer/transformer.dart" not found.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

*Added [Area-Pub](../labels/Area-Pub), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Pub doesn't explicitly handle any proxy configuration; it just calls dart:io and dart:isolate and lets those libraries do the low-level work.

The problem here is that pub starts a server that serves Dart code, then spawns an isolate that imports this Dart code. Apparently this isn't working when a proxy is active. I've filed issue dart-lang/sdk#15215 to track the issue in the underlying libraries.
______
*Marked this as being blocked by dart-lang/sdk#15215.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Issue dart-lang/sdk#15582 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/sgjesse"><img src="https://avatars.githubusercontent.com/u/1343914?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sgjesse](https://github.com/sgjesse)**

----

After looking further into this the 'no_proxy' setting in comment #­2 is wrong. The syntax 127.0.0.0/8 is not supported. Use 127.0.0.1 instead:

$ no_proxy="localhost,127.0.0.1,::1,[::1]" pub build

Should work.

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Removed [Priority-Unassigned](../labels/Priority-Unassigned) label.*
*Added [Priority-Medium](../labels/Priority-Medium) label.*

**Comment by DartBot** [comment]
*This comment was originally written by trevor.bo...&#064;gmail.com*
_____
I'm having the same issue with 'Transformer library "package:observe/transformer.dart" not found.', but I'm not behind a proxy.  My code works fine in dartium, run alone or with the dart-editor, but when I try to "pub build" on the command line, it fails with the aforementioned "not found" error.  Very confused, because the file is obviously  there.  Here's the output of my `pub build -v`: http://sprunge.us/AJRI

I've also attached it to this message.

There are some warnings in there.
______
**Attachment:**
[build.log](https://storage.googleapis.com/google-code-attachments/dart/issue-15161/comment-9/build.log) (136.80 KB)

**Comment by DartBot** [comment]
*This comment was originally written by trevor.borti...&#064;gmail.com*
_____
Oh, and for reference, here's my example project: https://github.com/wulftone/tabby-element/

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

@trevor, I'm not seeing any errors in either of those logs. Am I missing something?

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

This looks like it might be stale. Is anyone on this thread still having problems?

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

*Added [AssumedStale](../labels/AssumedStale) label.*

**Comment by DartBot** [comment]
*This comment was originally written by blit...&#064;gmail.com*
_____
hi i am seeing the same error on window
Pub build failed, [1] Transformer library "package:observe/transformer.dart" not found.

with Dart Editor version 1.2.0.release behind a proxy.

code works fine in dartium but cannont build and run as javascript.

**Comment by DartBot** [comment]
*This comment was originally written by marti...&#064;lundogbendsen.dk*
_____
Hi. I saw the same in dart 1.3...without a proxy...
pub serve
Loading source assets... (0.8s)
Loading di transformers... (0.8s)
Transformer library "package:di/di.dart" not found.

Turns out my pub cache (AppData\Roaming\Pub\Cache) was corrupt (because I once was behind a proxy), and removing it solved the problem.

**Comment by DartBot** [comment]
*This comment was originally written by berz...&#064;gmail.com*
_____
It's not stale. Still having some problems, probably related to that proxy issue.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

It sounds from issue dart-lang/sdk#15215 that the solution is to disable the proxy when using pub by setting the environment variable "no_proxy=localhost" when invoking pub.

**Comment by trevor** [comment]
@munificent :heart_eyes:
